azure-pipelines.yaml

```diff
@@ -3,50 +3,76 @@ name: Nodejs Chart CI Pipeline
 trigger:
   branches:
     include:
-    - refs/tags/*
+      - refs/tags/*
 
 pr:
   branches:
     include:
-    - master
+      - master
 
 resources:
   repositories:
-  - repository: cnp-library
-    type: github
-    name: hmcts/cnp-azuredevops-libraries
-    endpoint: hmcts
+    - repository: cnp-library
+      type: github
+      name: hmcts/cnp-azuredevops-libraries
+      endpoint: hmcts
 
 variables:
   - name: agentPool
     value: ubuntu-latest
+  - name: acrName
+    value: hmctspublic
+  - name: acrResourceGroup
+    value: rpe-acr-prod-rg
+  - name: serviceConnection
+    value: azurerm-prod
+  - name: buildPath
+    value: chart-nodejs/test-image
+  - name: repoName
+    value: chart-nodejs
+  - name: valuesFile
+    value: $(repoName)/ci-values.yaml
+  - name: testAppImage
+    value: hmctspublic.azurecr.io/hmcts/chart-nodejs
 
 jobs:
-- job: Validate
-  pool:
-    vmImage: ${{ variables.agentPool }}
-  steps:
-  - template: steps/charts/validate.yaml@cnp-library
-    parameters:
-      chartNamespace: chart-tests
-      chartName: nodejs
-      chartReleaseName: chart-nodejs-ci-test
-
-- job: Release
-  # Make sure we are on the master branch with a tag
-  # to trigger this job
-  condition: >
-    and(
-        succeeded(),
-        startsWith(variables['Build.SourceBranch'], 'refs/tags/')
-      )
-  dependsOn: Validate
-  pool:
-    vmImage: ${{ variables.agentPool }}
-  steps:
-  - template: steps/charts/release.yaml@cnp-library
-    parameters:
-      chartNamespace: chart-tests
-      chartName: nodejs
-      chartReleaseName: chart-nodejs
+  - job: Validate
+    pool:
+      vmImage: ${{ variables.agentPool }}
+    steps:
+      - checkout: self
+        clean: true
 
+      - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+          - template: steps/acr-build.yaml@cnp-library
+            parameters:
+              serviceConnection: $(serviceConnection)
+              buildPath: $(buildPath)
+
+      - template: steps/charts/validate.yaml@cnp-library
+        parameters:
+          chartNamespace: chart-tests
+          chartName: nodejs
+          chartReleaseName: chart-nodejs-ci-test
+          chartPath: $(repoName)/
+          valuesFile: $(valuesFile)
+          additionalHelmArgs: --set image=$(testAppImage):pr-$(System.PullRequest.PullRequestNumber)
```

Comment:
isn't this always adding a PR number? even on master?

Reply:
yep, fixed

Reply:
maybe not...

Reply:
no it is fixed